SonarQube.MSBuild.Tasks/AttachBuildWrapper.cs

```diff
@@ -163,14 +148,18 @@ private bool Attach()
 
             // Choose either the 32- or 64-bit version of the monitor to match the current process
             string exeName = (Environment.Is64BitProcess ? BuildWrapperExeName64 : BuildWrapperExeName32);
-            string monitorExeFilePath = Path.Combine(this.BuildWrapperBinaryDir, exeName);
+            string monitorExeFilePath = Path.Combine(this.BinDirectoryPath, exeName);
 
             ProcessRunnerArguments args = new ProcessRunnerArguments(monitorExeFilePath, new MSBuildLoggerAdapter(this.Log));
             args.TimeoutInMilliseconds = this.buildWrapperTimeoutInMs;
 
             // See https://jira.sonarsource.com/browse/CPP-1469 for the specification of the arguments to pass to the Cpp plugin
             // --msbuild - task < PID > < OUTPUT_DIR >
-            args.CmdLineArgs = new string[] { "--msbuild-task", currentPID, OutputDirectoryPath };
+            args.CmdLineArgs = new string[] {
+                "--msbuild-task",
+                currentPID,
+                Path.GetFullPath(OutputDirectoryPath) // make sure the path is in canonical form
```

Comment:
Minor tweak: make sure the path doesn't contain any relative sections or repeated \\ before we pass it to the build wrapper.